File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.java
class Solution {
    public List<Double> averageOfLevels(TreeNode root) {
        Queue<TreeNode> que = new ArrayDeque<>();
        List<Double> result = new ArrayList<>();

        que.add(root);

        while (!que.isEmpty()) {
            int n = que.size();
            double total = 0.0;

            for (int i = 0; i < n; i++) {
                TreeNode now = que.poll();
                total += now.val;
                if (now.left != null) {
                    que.add(now.left);
                }
                if (now.right != null) {
                    que.add(now.right);
                }
            }

            result.add(total / n);
        }

        return result;
    }
}

File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.cpp
class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        queue<TreeNode*> que;
        vector<double> result;

        que.push(root);

        while (!que.empty()) {
            int n = que.size();
            double total = 0.0;

            for (int i = 0; i < n; i++) {
                total += que.front()->val;
                if (que.front()->left != NULL) {
                    que.push(que.front()->left);
                }
                if (que.front()->right != NULL) {
                    que.push(que.front()->right);
                }
                que.pop();
            }
            result.push_back(total / n);
        }

        return result;
    }
};